### ISSUE TITLE
SortFilter and DateFilter are broken on Swagger UI site

### ISSUE THREAD

**Issue by ryanhowdy** [api-platform/api-platform#2790]
**API Platform version(s) affected**: 4.0.6

**Description**  
I'm using Laravel with the IsApiResource Trait for my example.

OrderFilter
`new QueryParameter(key: 'sort[:property]', filter: OrderFilter::class)` will not work because on the swagger ui site you do not get the option to set which property you are sorting by, just an option to set 'asc' or 'desc'.

DateFilter
`new QueryParameter(key: 'created', filter: DateFilter::class, property: 'created_at')` does not work on the swagger ui site because all you get is a box to enter the date value.  And whatever you put in there just gets turned into `KEY=INPUT`.

So if I enter `[lte]=2024-07-10` it gets turned into `created=[lte]=2024-07-10` when what I really need is `create[lte]=2024-07-10`.

**How to reproduce**  
1. Create a new laravel project with at least one table with a date.
2. Add the following code to the table's model
```
         return [    
             new ApiResource(
                 operations: [
                     new Get(),
                     new GetCollection(),
                 ],
             ),
             new QueryParameter(key: 'created', filter: DateFilter::class, property: 'created_at'),
         ];
```
3. Go to the swagger ui site and try and search for created records before a date.
4. Notice the actual api call that is run is malformed.

**Comment by jonericksonbkjg** [comment]
@ryanhowdy have you found a cure for this? Suffering all sorts of issues with this when running with Laravel.

**Comment by ryanhowdy** [comment]
No, i don't think the swagger specs allow for these type of complicated params.

**Comment by soyuka** [comment]
They do allow this: 

https://github.com/api-platform/core/blob/a8d0c8471060039bdd0b8d118dcf7f0e97709276/src/Doctrine/Orm/Filter/DateFilter.php#L286-L297

There must be a bug I'll investigate, transfering this issue to `api-platform/core`, thanks for the report!

@jonericksonbkjg please open a new issue.
